django/urls/resolvers.py

```diff
@@ -164,12 +169,11 @@ def _check_pattern_startswith_slash(self):
         """
         Check that the pattern does not begin with a forward slash.
         """
```

Comment:
Nice. 👍
I saw this taking some time in the asv profile.  

Reading your PR I learnt that performance isn't just the method being called (`startswith()` and `endswith()`) but the 'object' being accessed too.

Reply:
Yes, the `self.regex` was hiding the compile step as it went through the descriptor class(es).

Reply:
How did you access the asv profile btw? Is that from running it locally?

Reply:
Yes, I ran the benchmark locally with `asv profile` [1]. I also use the `gui=snakeviz` option and it opens the pretty chart in my browser. 

[1] https://asv.readthedocs.io/en/stable/using.html#running-a-benchmark-in-the-profiler